recursion.rb

```diff
@@ -1,14 +1,43 @@
-
+# RC: n! = n * (n-1)!
+# BC: 1! = 1
+# BC: 0! = 1
 def fact(n)
+  return 1 if n == 0
+  return n * fact(n-1)
+
 end
 
 def fib(n)
+  return 1 if n == 1 || n == 2
+  return fib(n-1) + fib(n-2)
 end
```

Comment:
Crystal and Jeremy keep modeling all their code after a Fibonacci sequence that starts with 1 1 2 3 5 ...which as a math teacher is a little odd , but I guess I could have added in some code to deal with users putting in bad data of any kind.